stock_location_orderpoint/models/stock_location_orderpoint.py

```diff
@@ -0,0 +1,416 @@
+        help="Moves created through this orderpoint "
+        "will be put in this procurement group. "
+        "If none is given, the moves generated by stock rules "
+        "will be grouped into one big picking.",
+    )
+    location_src_id = fields.Many2one(
+        "stock.location", compute="_compute_location_src_id", store=True
+    )
+    active = fields.Boolean("Active", default=True)
+    priority = fields.Selection(
+        PROCUREMENT_PRIORITIES,
+        "Priority",
+        default="0",
+    )
+
+    _sql_constraints = [
+        (
+            "location_route_unique",
+            "unique(location_id, route_id)",
+            "The combination of Company, Location and Route must be unique",
+        )
+    ]
+
+    @api.constrains("location_id", "route_id")
+    def _check_location_id_route_id(self):
+        for orderpoint in self:
+            if (
+                not orderpoint.route_id
+                or orderpoint.location_id in orderpoint.route_id.rule_ids.location_id
+            ):
+                continue
+            raise ValidationError(
+                _(
+                    "The selected route {} must contain "
+                    "a rule where the Destination Location is {}"
+                ).format(
+                    orderpoint.route_id.display_name,
+                    orderpoint.location_id.display_name,
+                )
+            )
+
+    def _get_location_src_id(self):
+        self.ensure_one()
+        values = {
+            "warehouse_id": self.location_id.get_closest_warehouse(),
+            "route_ids": self.route_id,
+        }
+        current_location = self.location_id
+        while current_location:
+            rule = self.env["procurement.group"]._get_rule(
+                self.env["product.product"], current_location, values
+            )
+            if rule.procure_method == "make_to_stock":
+                return rule.location_src_id
+            current_location = rule.location_src_id
+
+    @api.depends("location_id", "route_id")
+    def _compute_location_src_id(self):
+        for orderpoint in self:
+            orderpoint.location_src_id = orderpoint._get_location_src_id()
+
+    def _prepare_procurement(self, product, qty, date_planned, proc_vals):
+        self.ensure_one()
+        proc_vals = copy(proc_vals)
+        proc_vals.update(
+            {
+                "date_planned": date_planned or fields.Datetime.now(),
+            }
+        )
+        return self.env["procurement.group"].Procurement(
+            product,
+            qty,
+            product.uom_id,
+            self.location_id,
+            self.name,
+            self.name,
+            self.company_id,
+            proc_vals,
+        )
+
+    def _prepare_procurement_values(self):
+        self.ensure_one()
+        return {
+            "route_ids": self.route_id,
+            "date_deadline": False,
+            "warehouse_id": self.location_id.get_closest_warehouse(),
```

Comment:
@mt-software-de Could you add a TODO here to remind me to remove that for v16 as warehouse_id field on locations is correctly implemented ?

Reply:
No we cannot rely on odoo code for the computed warehouse because we have a virtual warehouse containing other warehouses:
WH/Stock (virtual warehouse for stock available to promise)
WH/Stock/MyWh1 (real warehouse1)
WH/Stock/MyWh2 (real warehouse2)
If the location to replenish is WH/Stock/MyWh1/Internal, then it must find the warehouse MyWh1 and not WH

Reply:
No i dont think we can remove it. 
Look at my answer Here. https://github.com/OCA/stock-logistics-warehouse/pull/1640#issuecomment-1451636719